java/src/apps/gui3/FirstTimeStartUpWizard.java

```diff
@@ -256,6 +261,23 @@ JPanel createButtonPanel() {
         return buttonPanel;
     }
 
+    @Override
+    public void uncaughtException(Thread t, Throwable e) {
+        showDialogue(e);
+    }
+
+    private void showDialogue(Throwable ex) {
+        log.error("Exception: ", ex);
+        Cursor normalCursor = new Cursor(Cursor.DEFAULT_CURSOR);
+        parent.setCursor(normalCursor);
+        JOptionPane.showMessageDialog(parent,
```

Comment:
Shouldn't the method `JOptionPane.showMessageDialog()` be called on the swing thread (the GUI thead)? It seems to me that the method `showDialogue()` will be called on another thread.

https://docs.oracle.com/javase/7/docs/api/javax/swing/JOptionPane.html
Note the comment:
Warning: Swing is not thread safe. For more information see [Swing's Threading Policy](https://docs.oracle.com/javase/7/docs/api/javax/swing/package-summary.html#threading).

Reply:
Thanks, good review.